generators/client/templates/react/src/main/webapp/swagger-ui/index.html.ejs

```diff
@@ -17,190 +17,65 @@
-                    window.swaggerUi.headerView.trigger('update-swagger-ui', {
-                        url: $('#select_baseUrl').val()
-                    });
-                    addApiKeyAuthorization();
-                });
-
-                function addApiKeyAuthorization() {
-                <%_ if (authenticationType === 'session' || authenticationType === 'uaa') { _%>
-                    var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("X-XSRF-TOKEN", getCSRF(), "header");
-                    window.swaggerUi.api.clientAuthorizations.add("key", apiKeyAuth);
-                <%_ } else if (authenticationType === 'oauth2') { _%>
-                    var authToken = JSON.parse(localStorage.getItem("jhi-authenticationToken")).access_token;
-                    var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("Authorization",
-                            "Bearer " + authToken, "header");
-                    window.swaggerUi.api.clientAuthorizations.add("key", apiKeyAuth);
-                <%_ } else if (authenticationType === 'jwt') { _%>
-                    var authToken = JSON.parse(localStorage.getItem("jhi-authenticationToken") || sessionStorage.getItem("jhi-authenticationToken"));
-                    var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("Authorization", "Bearer " + authToken, "header");
-                    window.swaggerUi.api.clientAuthorizations.add("jwt", apiKeyAuth);
-                <%_ } _%>
-                }
+<script type="text/javascript">
+    window.onload = function () {
+        var urls = [];
+        axios.get("/swagger-resources").then(function (response) {
+            response.data.forEach(function (resource) {
+                urls.push({"name": resource.name, "url": resource.location});
+            });
 
-                <%_ if (authenticationType === 'session' || authenticationType === 'oauth2' || authenticationType === 'uaa') { _%>
-                function getCSRF() {
-                    var name = "XSRF-TOKEN=";
-                    var ca = document.cookie.split(';');
-                    for(var i=0; i<ca.length; i++) {
-                        var c = ca[i];
-                        while (c.charAt(0)===' ') c = c.substring(1);
-                        if (c.indexOf(name) !== -1) return c.substring(name.length,c.length);
+            // Build a system
+            var ui = SwaggerUIBundle({
+                urls: urls,
+                dom_id: '#swagger-ui',
+                deepLinking: true,
+                filter: true,
+                layout: "StandaloneLayout",
+                withCredentials: true,
+                presets: [
+                    SwaggerUIBundle.presets.apis,
+                    SwaggerUIStandalonePreset
+                ],
+                plugins: [
+                    SwaggerUIBundle.plugins.DownloadUrl
+                ],
+                <%_ if (authenticationType === 'jwt') { _%>
+                onComplete: function () {
```

Comment:
it seems to me that jwt is the only case where the token has to be passed via js to the UI. In case of oauth and uaa the authentication tokens and session info are all in the cookies. am I right?